registration.go

```diff
@@ -0,0 +1,200 @@
+package notificationhubs
+
+import (
+	"bytes"
+	"context"
+	"encoding/xml"
+	"errors"
+	"path"
+	"strings"
+	"time"
+)
+
+// newRegistration initializes and returns a Notification pointer
+func newRegistration(deviceID string, expirationTime *time.Time, notificationFormat NotificationFormat,
+	registrationID string, tags string) *Registration {
+	return &Registration{
+		deviceID,
+		expirationTime,
+		notificationFormat,
+		registrationID,
+		tags,
+	}
+}
+
+// newTemplateRegistration initializes and returns a TemplateNotification pointer
+func newTemplateRegistration(deviceID string, expirationTime *time.Time, registrationID string, tags string,
+	platform TargetPlatform, template string) *TemplateRegistration {
+	return &TemplateRegistration{
+		deviceID,
+		expirationTime,
+		registrationID,
+		tags,
+		platform,
+		template,
+	}
+}
+
+// Normalize normalizes all devices in the feed
+func (r *Registrations) normalize() {
+	for _, entry := range r.Entries {
+		if entry.RegistrationContent != nil {
+			entry.RegistrationContent.normalize()
+		}
+	}
+}
+
+// Normalize the registration result
+func (r *RegistrationResult) normalize() {
+	if r.RegistrationContent != nil {
+		r.RegistrationContent.normalize()
+	}
+}
+
+// Normalize normalizes the different devices
+func (r *RegistrationContent) normalize() {
```

Comment:
Very big function 🤯 is it possible to break out some of the code that is in the lowest depth?